lib/extensions/as_dependencies_interlock.rb

```diff
@@ -0,0 +1,20 @@
+class AsDependenciesInterlock
+  # The classic autoloader needs the ActiveSupport::Dependencies.interlock
+  # to avoid deadlocks.  With zeitwerk, this actually causes deadlocks so
+  # we only use the interlock with the classic autoloader.
+  def self.loading(&block)
+    if Rails.application.config.autoloader == :zeitwerk
+      yield
+    else
+      ActiveSupport::Dependencies.interlock.loading(&block)
+    end
+  end
+
+  def self.permit_concurrent_loads(&block)
+    if Rails.application.config.autoloader == :zeitwerk
+      yield
+    else
+      ActiveSupport::Dependencies.interlock.permit_concurrent_loads(&block)
```

Comment:
heh. monkey patch AS::Dep.interlock to just yield if zeitwerk 😈

Reply:
ooh. Jason suggested this 33 minutes ago.

I wasn't being serious... but maybe?